python/torch_mlir/csrc/base_lazy_backend/mlir_lowering_context.cpp

```diff
@@ -108,17 +110,25 @@ void TorchMlirLoweringContext::AddParameter(
 ComputationPtr TorchMlirLoweringContext::Build() {
   PRINT_FUNCTION();
 
+  // Since we mutated the types of some nodes to insert shape information, we
```

Comment:
You might want to check if the static information adjustment I added in https://github.com/llvm/torch-mlir/pull/824 fixes this issue for you. Not opposed to refining the types, but just to know whether it is truly necessary.

Reply:
Hm I'm looking at some of the tests [here](https://github.com/silvasean/torch-mlir/blob/b0ad9b21e716f6291e4fb15d8795d14b982b9075/test/python/importer/jit_ir/node_import/tuple.py#L67-L68) and I'm not entirely sure this is what we want?
```
# CHECK-LABEL:   func @__torch__.tuple_construct_arg_needs_refinement(
# CHECK-SAME:                                                         %[[T0:.*]]: !torch.tensor,
# CHECK-SAME:                                                         %[[T1:.*]]: !torch.tensor) -> !torch.tuple<tensor, tensor> {
# CHECK:           %[[T0_REFINED:.*]] = torch.tensor_static_info_cast %[[T1]] : !torch.tensor to !torch.tensor<[4],f32>
# CHECK:           %[[T1_REFINED:.*]] = torch.tensor_static_info_cast %[[T0]] : !torch.tensor to !torch.tensor<[3],f32>
# CHECK:           %[[TUPLE:.*]] = torch.prim.TupleConstruct %[[T0_REFINED]], %[[T1_REFINED]] : !torch.tensor<[4],f32>, !torch.tensor<[3],f32> -> !torch.tuple<tensor<[4],f32>, tensor<[3],f32>>
# CHECK:           %[[DEREFINED:.*]] = torch.derefine %[[TUPLE]] : !torch.tuple<tensor<[4],f32>, tensor<[3],f32>> to !torch.tuple<tensor, tensor>
# CHECK:           return %[[DEREFINED]] : !torch.tuple<tensor, tensor>
```

For our use case, we want to directly change the output type of `TupleConstruct` to reflect the updated one, because otherwise the type is just completely wrong. 

e.g. Before:
```
graph(%a : Float(123), %b : Float(4, 5, 6)):
    c : (Tensor, Tensor) = prim::TupleConstruct(%a, %b)
    return (%c)
```
After:
```
graph(%a : Float(123), %b : Float(4, 5, 6)):
    c : (Float(123), Float(4, 5, 6)) = prim::TupleConstruct(%a, %b)
    return (%c)
```
It looks like your solution does a cast operation before and after, which I feel is unnecessary to solve this issue.